src/echo/echo.rs

```diff
@@ -36,119 +34,82 @@ static HELP: &str = r#"
  \\xHH    byte with hexadecimal value HH (1 to 2 digits)
 "#;
 
-enum Base {
-    B8 = 8,
-    B16 = 16,
-}
-
-struct Opts {
-    newline: bool,
-    escape: bool,
+pub fn parse_code(input: &mut std::iter::Peekable<std::str::Chars<'_>>, base: u32, max_digits: u32, bits_per_digit: u32) -> char {
+    let mut ret = 0;
+    for _ in 0..max_digits {
+        match input.peek() {
+            Some(c) => match c.to_digit(base) {
+                Some(n) => ret = (ret << bits_per_digit) | n,
+                None => break,
+            },
+            None => break,
+        }
+        input.next();
+    }
+    std::char::from_u32(ret).unwrap()
```

Comment:
This is still not quite correct.  If no valid hex or octal characters follow the `\x` or `\0` (respectively), then `\x` or `\0` should be printed out as-is (just like if they were not escape sequences).

Reply:
I see. I wasn't sure if there was well-defined behavior for malformed input. Simple fix.